tests/generic_config_updater/test_mgmt_interface.py

```diff
@@ -0,0 +1,126 @@
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+def test_forced_mgmt_routes_update(duthost, ensure_dut_readiness):
+    # Get interface and check config generate correct
+    mgmt_interface_keys = duthost.command("sonic-db-cli  CONFIG_DB keys 'MGMT_INTERFACE|eth0|*'")['stdout']
+    logging.debug("mgmt_interface_keys: {}".format(mgmt_interface_keys))
+
+    for interface_key in mgmt_interface_keys.split('\n'):
+        logging.debug("interface_key: {}".format(interface_key))
+        interface_address = interface_key.split('|')[2]
+
+        # Get current forced mgmt routes
+        forced_mgmt_routes_config = duthost.command("sonic-db-cli CONFIG_DB HGET '{}' forced_mgmt_routes@"
+                                                    .format(interface_key))['stdout']
+
+        original_forced_mgmt_routes = []
+        if forced_mgmt_routes_config != "":
+            original_forced_mgmt_routes = forced_mgmt_routes_config.split(",")
+
+        # Prepare new forced mgmt routes
+        test_route = "1::2:3:4/64"
+        ip_type = "-6"
+        if type(ipaddress.ip_network(interface_address, False)) == ipaddress.IPv4Network:
+            test_route = "1.2.3.4/24"
+            ip_type = "-4"
+
+        updated_forced_mgmt_routes = original_forced_mgmt_routes.copy()
+        updated_forced_mgmt_routes.append(test_route)
+        logging.debug("interface address: {}, original_forced_mgmt_routes: {}, updated_forced_mgmt_routes: {}"
+                      .format(interface_address, original_forced_mgmt_routes, updated_forced_mgmt_routes))
+
+        def update_forced_mgmt_route(duthost, routes):
+            # Escape '/' in interface key
+            json_patch = [
+                {
+                    "path": "/MGMT_INTERFACE/eth0|{}/forced_mgmt_routes".format(interface_address.replace("/", "~1"))
+                }
+            ]
+
+            if len(routes) == 0:
+                json_patch[0]["op"] = "remove"
+            else:
+                json_patch[0]["value"] = routes
+                # Replace if forced_mgmt_routes already exist
+                current_config = duthost.command("sonic-db-cli CONFIG_DB HGET '{}' forced_mgmt_routes@"
+                                                 .format(interface_key))['stdout']
+                if current_config != "":
+                    json_patch[0]["op"] = "replace"
+                else:
+                    json_patch[0]["op"] = "add"
+
+            tmpfile = generate_tmpfile(duthost)
+            try:
+                output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+                logging.debug("json_patch: {}".format(json_patch))
+                logging.debug("apply_patch result: {}".format(output))
+            finally:
+                delete_tmpfile(duthost, tmpfile)
+
+        # Update forced mgmt routes with new route address
+        wait_for_file_changed(
```

Comment:
Seems could improve code with line 112 to avoid duplicates

Reply:
Fixed